src/share/classes/com/sun/javatest/regtest/exec/RegressionScript.java

```diff
@@ -1175,18 +1176,33 @@ Map<String, String> getTestProperties()  {
         p.put("test.root", getTestRootDir().getPath());
         return Collections.unmodifiableMap(p);
     }
-    // where
+
     private String toString(List<Path> files) {
         return files.stream()
                 .map(Path::toString)
                 .collect(Collectors.joining(File.pathSeparator));
     }
-//    // where
-//    private String toString(List<File> files) {
-//        return files.stream()
-//                .map(File::getPath)
-//                .collect(Collectors.joining(File.pathSeparator));
-//    }
+
+    private Optional<String> computeMainClassNameFromTestFile() {
+        Path file = locations.absTestFile();
+        // Take file name as basis for class name.
+        String fileName = file.getFileName().toString();
+        int fileNameLength = fileName.length(); // including ".java".length() = 5
+        if (fileNameLength <= 5 && !fileName.endsWith(".java")) {
```

Comment:
This logic seems suspicious to me. If `fileNameLength` will be over `5` characters, the extension will be ignored. If the extension is `.java`, the length will be ignored. Should it be:
```suggestion
        if (fileNameLength <= 5 || !fileName.endsWith(".java")) {
```